modules/nf-core/mafft/main.nf

```diff
@@ -20,8 +20,10 @@ process MAFFT {
 
     script:
     def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: '--add'
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def add = addsequences ? "--add $addsequences" : ''
+    def add = addsequences ? "${args2} ${addsequences}" : ''
```

Comment:
The way it was done before was correct. If you don't want to use the `addsequences` files, you simply don't enter it into the module.

Reply:
I addition to that: `args2` should only be used for the arguments of a secondary tool used by the module and not for logic in the module :)

Reply:
https://github.com/nf-core/modules/issues/4353, there are multiple ways to add sequences https://mafft.cbrc.jp/alignment/server/add.html. 
It's not possible to use those other methods the way it was written previously.

Reply:
IMO the best option would be to either that input a tuple with the file and a value that states which input it should be or to create an input channel for each option. The last option is the most nf-core-y way of doing it though :grin: (but I like the first one more)

Reply:
Oh yes if it's possible to do combinations of them all, you should definitely make them separate input channels (sorry, I know it seems weird :grin: ).